File: SSSExampleVBnet/SSSExampleVBnet/SSSExampleVBnet/Form1.vb
Imports ScreenScraper

Enum CaptureMethod
    NATIVE
    FULL_TEXT
    OCR
End Enum

Public Class Form1
    Private m_tCapture As TextCaptureX
    Private m_tFulltextCapt As GetAAText
    Private m_tOCRCapture As GetOCRText
    Private m_tSelection As TSelection
    Private m_bRegOrWnd As Boolean
    Private m_bCanTryAgain As Boolean
    Private Function GetCaptureMethod() As CaptureMethod
        If (radioNative.Checked = True) Then
            GetCaptureMethod = CaptureMethod.NATIVE
        ElseIf (radioFullText.Checked = True) Then
            GetCaptureMethod = CaptureMethod.FULL_TEXT
        Else
            GetCaptureMethod = CaptureMethod.OCR
        End If
    End Function
    Private Sub btnCollect_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles btnCollect.Click
        m_bRegOrWnd = True
        Dim tSelInfo As TSelectionInfo

        Try
            tSelInfo = m_tSelection.Start(TS_SELECTION.tsSelectionRectangle, TS_SELECTION_OPTIONS.tsSelFlagDefaultText)
        Catch ex As Exception
            MessageBox.Show(ex.Message)
            Return
        End Try

        'get the hwnd
        Dim hwnd As Integer
        hwnd = tSelInfo.WindowHandle
        txtHandle.Text = hwnd.ToString("X")

        Dim uiElement As UIElem
        uiElement = ComFactory.Instance.NewUIElem
        uiElement.hwnd = hwnd

        'set the UI element ID
        textBox1.Text = uiElement.GetID(True)

        'get the selection points
        tSelInfo.GetClientCoordinates()

        'set the points
        txtX.Text = tSelInfo.RCLeft.ToString
        txtY.Text = tSelInfo.RCTop.ToString
        txtWidth.Text = tSelInfo.RCWidth.ToString
        txtHeight.Text = tSelInfo.RCHeight.ToString
    End Sub

    Private Sub Form1_Load(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles MyBase.Load
        Try
            m_tCapture = ComFactory.Instance.NewTextCaptureX
            m_tFulltextCapt = ComFactory.Instance.NewGetAAText
            m_tOCRCapture = ComFactory.Instance.NewGetOCRText
            m_tSelection = ComFactory.Instance.NewTSelection

            m_tCapture.UseClientCoordinates = True
            m_tOCRCapture.UseClientCoordinates = True
            m_tFulltextCapt.UseClientCoordinates = True

        Catch ex As Exception
            MessageBox.Show(ex.Message)
        End Try
        'init ui 
        radioNative.Checked = True
        checkIDorHWND.Checked = True

        'init ocr
        comboEngine.Items.Add("Tesseract")
        comboEngine.SelectedIndex = 0

        comboLang.Items.Add("eng")
        comboLang.Items.Add("deu")
        comboLang.Items.Add("spa")
        comboLang.Items.Add("ita")
        comboLang.Items.Add("fra")
        comboLang.Items.Add("nld")
        comboLang.Items.Add("spa")
        comboLang.SelectedIndex = 0
    End Sub

    Private Sub button1_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles button1.Click
        m_bRegOrWnd = False
        Dim tSelInfo As TSelectionInfo

        WindowState = FormWindowState.Minimized

        Try
            tSelInfo = m_tSelection.Start(TS_SELECTION.tsSelectionUIElement, TS_SELECTION_OPTIONS.tsSelFlagDefaultText)
        Catch ex As Exception
            MessageBox.Show(ex.Message)
            Return
        End Try

        WindowState = FormWindowState.Normal

        'get the hwnd
        Dim hwnd As Integer
        hwnd = tSelInfo.WindowHandle
        txtHandle.Text = hwnd.ToString("X")

        'set the UI elemend ID
        Dim uiElement As UIElem
        uiElement = ComFactory.Instance.NewUIElem

        If (tSelInfo.UIElementID = "") Then
            textBox1.Text = "A valid ID could not be generated!"
            Exit Sub
        Else

            uiElement.InitializeFromID(tSelInfo.UIElementID, True)
            Dim uiElemID As String
            uiElemID = uiElement.GetID(True)
            textBox1.Text = uiElemID

        End If


    End Sub
    Private Sub PerformRegionCapture(ByVal hwnd As Integer, ByVal x As Integer, ByVal y As Integer, ByVal width As Integer, ByVal height As Integer)
        If (m_bRegOrWnd = False) Then
            'if a window was selected
            MessageBox.Show("You selected a window!")
            Return
        End If

        Dim currentMethod As CaptureMethod
        currentMethod = GetCaptureMethod()
        Dim strText As String
        strText = ""

        If (currentMethod = CaptureMethod.NATIVE) Then
            Dim font As stdole.IFontDisp

            Try
                'preserve layout checkbox
                If (checkLayout.Checked = True) Then
                    m_tCapture.FormattedText = True
                End If

                strText = m_tCapture.GetTextFromRectWithFont(hwnd, x, y, width, height, font)
                txtFont.Text = font.Name
                txtFontSize.Text = font.Size.ToString
            Catch ex As Exception
                Dim ComEx As System.Runtime.InteropServices.COMException
                ComEx = ex
                If (ComEx.ErrorCode = -2147220989) And (m_bCanTryAgain = True) Then
                    m_bCanTryAgain = False
                    MessageBox.Show("Capture timeout! Trying again!")
                    System.Threading.Thread.Sleep(4000) 'sleep for 4 seconds
                    PerformRegionCapture(hwnd, x, y, width, height)
                    Return
                Else
                    strText = "Capture Error!"
                    MessageBox.Show(ex.Message)
                End If

            End Try
        ElseIf (currentMethod = CaptureMethod.FULL_TEXT) Then
            Try
                strText = m_tFulltextCapt.GetFullTextFromRectangle(hwnd, x, y, width, height)
                txtFont.Text = ""
            Catch ex As Exception
                Dim ComEx As System.Runtime.InteropServices.COMException
                ComEx = ex
                If (ComEx.ErrorCode = -2147220989) And (m_bCanTryAgain = True) Then
                    m_bCanTryAgain = False
                    MessageBox.Show("Capture timeout! Trying again!")
                    System.Threading.Thread.Sleep(4000) 'sleep for 4 seconds
                    PerformRegionCapture(hwnd, x, y, width, height)
                    Return
                Else
                    strText = "Capture Error!"
                    MessageBox.Show(ex.Message)
                End If
            End Try
        ElseIf (currentMethod = CaptureMethod.OCR) Then
            Try
                Dim strLang As String
                strLang = comboLang.Items(comboLang.SelectedIndex)
                strText = m_tOCRCapture.GetTextFromRect(hwnd, x, y, width, height, strLang, False)
                txtFont.Text = ""
            Catch ex As Exception
                Dim ComEx As System.Runtime.InteropServices.COMException
                ComEx = ex
                If (ComEx.ErrorCode = -2147220989) And (m_bCanTryAgain = True) Then
                    m_bCanTryAgain = False
                    MessageBox.Show("Capture timeout! Trying again!")
                    System.Threading.Thread.Sleep(4000) 'sleep for 4 seconds
                    PerformRegionCapture(hwnd, x, y, width, height)
                    Return
                Else
                    strText = "Capture Error!"
                    MessageBox.Show(ex.Message)
                End If
            End Try

        End If
        txtResult.Text = strText
    End Sub

    Private Sub btnRectangle_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles btnRectangle.Click
        Dim x, y, width, height As Integer
        x = Integer.Parse(txtX.Text)
        y = Integer.Parse(txtY.Text)
        width = Integer.Parse(txtWidth.Text)
        height = Integer.Parse(txtHeight.Text)

        Dim uiElement As UIElem
        Try
            uiElement = ComFactory.Instance.NewUIElem

        Catch ex As Exception
            MessageBox.Show(ex.Message)
            Return
        End Try

        If (checkIDorHWND.Checked = True) Then
            Dim strId As String
            strId = textBox1.Text
            Try
                uiElement.InitializeFromID(strId, True)
            Catch ex As Exception
                MessageBox.Show(ex.Message)
                Return
            End Try
        Else
            Dim hwnd As Integer
            hwnd = Integer.Parse(txtHandle.Text, System.Globalization.NumberStyles.HexNumber)
            uiElement.hwnd = hwnd
        End If

        m_bCanTryAgain = True
        PerformRegionCapture(uiElement.hwnd, x, y, width, height)
    End Sub

    Private Sub button2_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles button2.Click
        Dim uiElement As UIElem
        Try
            uiElement = ComFactory.Instance.NewUIElem
        Catch ex As Exception
            MessageBox.Show(ex.Message)
        End Try
        If (checkIDorHWND.Checked = True) Then
            'get id
            Dim strID As String
            strID = textBox1.Text
            Try
                uiElement.InitializeFromID(strID, True)
            Catch ex As Exception
                MessageBox.Show(ex.Message)
            End Try
        Else
            Dim hwnd As Integer
            hwnd = Integer.Parse(txtHandle.Text, System.Globalization.NumberStyles.HexNumber)
            uiElement.hwnd = hwnd
        End If

        m_bCanTryAgain = True
        PerformWndCapture(uiElement)
    End Sub
    Private Sub PerformWndCapture(ByVal uiElement As UIElem)
        If (m_bRegOrWnd = True) Then
            MessageBox.Show("You selected a region!")
            Return
        End If

        Dim font As stdole.IFontDisp
        Dim strText As String
        strText = ""

        Dim captMethod As CaptureMethod
        captMethod = GetCaptureMethod()
        If (captMethod = CaptureMethod.NATIVE) Then
            Try
                'preserve layout checkbox
                If (checkLayout.Checked = True) Then
                    m_tCapture.FormattedText = True
                End If
                strText = m_tCapture.GetTextFromUIElemWithFont(uiElement, font)
                txtFont.Text = font.Name
                txtFontSize.Text = font.Size.ToString
            Catch ex As Exception
                Dim ComEx As System.Runtime.InteropServices.COMException
                ComEx = ex
                If (ComEx.ErrorCode = -2147220989) And (m_bCanTryAgain = True) Then
                    m_bCanTryAgain = False
                    MessageBox.Show("Capture timeout! Trying again!")
                    System.Threading.Thread.Sleep(4000) 'sleep for 4 seconds
                    PerformWndCapture(uiElement)
                    Return
                Else
                    strText = "Capture Error!"
                    MessageBox.Show(ex.Message)
                End If
            End Try
        ElseIf (captMethod = CaptureMethod.FULL_TEXT) Then
            Try
                strText = m_tFulltextCapt.GetFullTextFromUIElem(uiElement)
                txtFont.Text = ""
            Catch ex As Exception
                Dim ComEx As System.Runtime.InteropServices.COMException
                ComEx = ex
                If (ComEx.ErrorCode = -2147220989) And (m_bCanTryAgain = True) Then
                    m_bCanTryAgain = False
                    MessageBox.Show("Capture timeout! Trying again!")
                    System.Threading.Thread.Sleep(4000) 'sleep for 4 seconds
                    PerformWndCapture(uiElement)
                    Return
                Else
                    strText = "Capture Error!"
                    MessageBox.Show(ex.Message)
                End If
            End Try
        ElseIf (captMethod = CaptureMethod.OCR) Then
            Try
                Dim strlang As String
                strlang = comboLang.Items(comboLang.SelectedIndex)
                strText = m_tOCRCapture.GetTextFromUIElem(uiElement, strlang, False)
                txtFont.Text = ""
            Catch ex As Exception
                Dim ComEx As System.Runtime.InteropServices.COMException
                ComEx = ex
                If (ComEx.ErrorCode = -2147220989) And (m_bCanTryAgain = True) Then
                    m_bCanTryAgain = False
                    MessageBox.Show("Capture timeout! Trying again!")
                    System.Threading.Thread.Sleep(4000) 'sleep for 4 seconds
                    PerformWndCapture(uiElement)
                    Return
                Else
                    strText = "Capture Error!"
                    MessageBox.Show(ex.Message)
                End If
            End Try
        End If
        txtResult.Text = strText
    End Sub

    Private Sub btnActive_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles btnActive.Click
        Dim activeHwnd As Integer
        WindowState = FormWindowState.Minimized
        Try
            activeHwnd = m_tCapture.GetActiveWindowHwnd()
            Dim strResult As String
            strResult = m_tCapture.CaptureWindow(activeHwnd)
            txtResult.Text = strResult
        Catch ex As Exception
            MessageBox.Show(ex.Message)
            txtResult.Text = "Capture Error!"
        End Try
        WindowState = FormWindowState.Normal
    End Sub

    Private Sub btnPoint_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles btnPoint.Click
        Dim x, y, hwnd As Integer
        x = Integer.Parse(txtX.Text)
        y = Integer.Parse(txtY.Text)
        hwnd = Integer.Parse(txtHandle.Text, System.Globalization.NumberStyles.HexNumber)

        Try
            Dim strResult As String
            strResult = m_tCapture.GetTextFromPoint(hwnd, x, y)
            txtResult.Text = strResult
        Catch ex As Exception
            MessageBox.Show(ex.Message)
            txtResult.Text = "Capture Error!"
        End Try
    End Sub
End Class

File: _site/Samples/Legacy/SSSExampleCS/SSSExampleCS/SSSExampleCS/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Windows.Forms;
using ScreenScraper;

namespace SSSExampleCS
{
    enum CaptureMethod
    {
        NATIVE,
        FULL_TEXT,
        OCR
    };

    public partial class Form1 : Form
    {
        TextCaptureX m_tCapture;
        GetAAText m_tFullTextCapt;
        TSelection m_tSelection;
        GetOCRText m_tOCRCapture;
        bool m_bRegOrWnd;
        bool m_bCanTryAgain;

        public Form1()
        {
            InitializeComponent();

            try
            {
                m_tCapture = ComFactory.Instance.NewTextCaptureX();
                m_tSelection = ComFactory.Instance.NewTSelection();
                m_tFullTextCapt = ComFactory.Instance.NewGetAAText();
                m_tOCRCapture = ComFactory.Instance.NewGetOCRText();

                m_tCapture.UseClientCoordinates = true;
                m_tFullTextCapt.UseClientCoordinates = true;
                m_tOCRCapture.UseClientCoordinates = true;
            }
            catch (Exception ex)
            {
                MessageBox.Show(ex.Message);
            }

            // init radios
            radioNative.Checked = true;       
 
            // init checkbox
            checkIDorHWND.Checked = true;

            // init OCR
            comboEngine.Items.Add("Tesseract");
            comboEngine.SelectedIndex = 0;

            comboLang.Items.Add("eng");
            comboLang.Items.Add("deu");
            comboLang.Items.Add("ita");
            comboLang.Items.Add("fra");
            comboLang.Items.Add("nld");
            comboLang.Items.Add("spa");
            comboLang.SelectedIndex = 0;
        }

        private CaptureMethod GetCaptureMethod()
        {
            if (radioNative.Checked)
            {
                return CaptureMethod.NATIVE;
            }
            else if (radioFullText.Checked)
            {
                return CaptureMethod.FULL_TEXT;
            }
            return CaptureMethod.OCR;
        }

        private void PerformRegionCapture(int hwnd, int x, int y, int width, int height)
        {
            if (m_bRegOrWnd == false)
            {
                // if a window was selected
                MessageBox.Show("You selected a window!");
                return;
            }

            CaptureMethod currentMethod = GetCaptureMethod();
            string strText = "";
            if (currentMethod == CaptureMethod.NATIVE)
            {
                //stdole.IFontDisp font;
                try
                {
                    // preserve layout checkbox
                    if (checkLayout.Checked == true)
                    {
                        m_tCapture.FormattedText = true;
                    }
                    strText = m_tCapture.GetTextFromRect(hwnd, x, y, width, height);
                    //txtFont.Text = font.Name;
                    //txtFontSize.Text = font.Size.ToString();
                }
                catch (Exception ex)
                {
                    System.Runtime.InteropServices.COMException COMEx = (System.Runtime.InteropServices.COMException)ex;
                    if (COMEx.ErrorCode == -2147220989 && m_bCanTryAgain)
                    {
                        // this is the timeout exception

                        m_bCanTryAgain = false;
                        MessageBox.Show("Trying again!");
                        System.Threading.Thread.Sleep(4000); // sleep for 4 seconds
                        PerformRegionCapture(hwnd, x, y, width, height); // and try again

                        return;
                    }
                    else
                    {
                        strText = "Capture Error!";
                        MessageBox.Show(ex.Message);
                    }
                }
            }
            else if (currentMethod == CaptureMethod.FULL_TEXT)
            {
                try
                {
                    strText = m_tFullTextCapt.GetFullTextFromRectangle(hwnd, x, y, width, height);
                    txtFont.Text = "";
                }
                catch (Exception ex)
                {
                    System.Runtime.InteropServices.COMException COMEx = (System.Runtime.InteropServices.COMException)ex;
                    if (COMEx.ErrorCode == -2147220989 && m_bCanTryAgain)
                    {
                        // this is the timeout exception

                        m_bCanTryAgain = false;
                        MessageBox.Show("Trying again!");
                        System.Threading.Thread.Sleep(5000); // sleep for 4 seconds
                        PerformRegionCapture(hwnd, x, y, width, height); // and try again

                        return;
                    }
                    else
                    {
                        strText = "Capture Error!";
                        MessageBox.Show(ex.Message);
                    }
                }
            }
            else if (currentMethod == CaptureMethod.OCR)
            {
                try
                {                 
                    string strLang = (string)comboLang.Items[comboLang.SelectedIndex];
                    strText = m_tOCRCapture.GetTextFromRect(hwnd, x, y, width, height, strLang, false);
                    txtFont.Text = "";
                }
                catch (Exception ex)
                {
                    System.Runtime.InteropServices.COMException COMEx = (System.Runtime.InteropServices.COMException)ex;
                    if (COMEx.ErrorCode == -2147220989 && m_bCanTryAgain)
                    {
                        // this is the timeout exception

                        m_bCanTryAgain = false;
                        MessageBox.Show("Trying again!");
                        System.Threading.Thread.Sleep(5000); // sleep for 4 seconds
                        PerformRegionCapture(hwnd, x, y, width, height); // and try again

                        return;
                    }
                    else
                    {
                        strText = "Capture Error!";
                        MessageBox.Show(ex.Message);
                    }
                }
            }
            txtResult.Text = strText;
        }

        private void PerformWndCapture(UIElem uiElem)
        {
            if (m_bRegOrWnd == true)
            {
                // if a region was selected
                MessageBox.Show("You selected a region!");
                return;
            }

            if (uiElem == null)
            {
                return;
            }

            // get the text and font
            //stdole.IFontDisp font;
            string strText = "";
            CaptureMethod captMethod = GetCaptureMethod();
            if (captMethod == CaptureMethod.NATIVE)
            {
                try
                {
                    // preserve layout checkbox
                    if (checkLayout.Checked == true)
                    {
                        m_tCapture.FormattedText = true;
                    }
                    strText = m_tCapture.GetTextFromUIElem(uiElem);
                    //txtFont.Text = font.Name;
                    //txtFontSize.Text = font.Size.ToString();
                }
                catch (Exception ex)
                {
                    System.Runtime.InteropServices.COMException COMEx = (System.Runtime.InteropServices.COMException)ex;
                    if (COMEx.ErrorCode == -2147220989 && m_bCanTryAgain)
                    {
                        // this is the timeout exception

                        m_bCanTryAgain = false;
                        MessageBox.Show("Trying again!");
                        System.Threading.Thread.Sleep(5000); // sleep for 4 seconds
                        PerformWndCapture(uiElem); // and try again

                        return;
                    }
                    else
                    {
                        strText = "Capture Error!";
                        MessageBox.Show(ex.Message);
                    }
                }
            }
            else if (captMethod == CaptureMethod.FULL_TEXT)
            {
                try
                {
                    strText = m_tFullTextCapt.GetFullTextFromUIElem(uiElem);
                    txtFont.Text = "";
                }
                catch (Exception ex)
                {
                    System.Runtime.InteropServices.COMException COMEx = (System.Runtime.InteropServices.COMException)ex;
                    if (COMEx.ErrorCode == -2147220989 && m_bCanTryAgain)
                    {
                        // this is the timeout exception

                        m_bCanTryAgain = false;
                        MessageBox.Show("Trying again!");
                        System.Threading.Thread.Sleep(5000); // sleep for 4 seconds
                        PerformWndCapture(uiElem); // and try again

                        return;
                    }
                    else
                    {
                        strText = "Capture Error!";
                        MessageBox.Show(ex.Message);
                    }
                }
            }
            else if (captMethod == CaptureMethod.OCR)
            {
                try
                {
                    string strLang = (string)comboLang.Items[comboLang.SelectedIndex];
                    
                    strText = m_tOCRCapture.GetTextFromUIElem(uiElem, strLang, false);
                    txtFont.Text = "";
                }
                catch (Exception ex)
                {
                    System.Runtime.InteropServices.COMException COMEx = (System.Runtime.InteropServices.COMException)ex;
                    if (COMEx.ErrorCode == -2147220989 && m_bCanTryAgain)
                    {
                        // this is the timeout exception

                        m_bCanTryAgain = false;
                        MessageBox.Show("Trying again!");
                        System.Threading.Thread.Sleep(5000); // sleep for 4 seconds
                        PerformWndCapture(uiElem); // and try again

                        return;
                    }
                    else
                    {
                        strText = "Capture Error!";
                        MessageBox.Show(ex.Message);
                    }
                }
            }

            txtResult.Text = strText;
        }

        private void btnCollect_Click(object sender, EventArgs e)
        {
            m_bRegOrWnd = true;
            this.WindowState = FormWindowState.Minimized;
            System.Threading.Thread.Sleep(100);

            TSelectionInfo tSelInfo = null;

            try
            {
                tSelInfo = (TSelectionInfo)m_tSelection.Start(
                            (int)TS_SELECTION.tsSelectionRectangle,
                            (int)TS_SELECTION_OPTIONS.tsSelFlagDefaultText
                        );
            }
            catch (Exception ex)
            {
                MessageBox.Show(ex.Message);
                return;
            }

            // get the hwnd
            int hwnd = tSelInfo.WindowHandle;
            txtHandle.Text = hwnd.ToString("X");

            UIElem uiElem = ComFactory.Instance.NewUIElem();
            uiElem.hwnd = hwnd;

            // set the UI element ID
            textBox1.Text = uiElem.GetID(true);

            // get the selection points
            tSelInfo.GetClientCoordinates();


            // set the points
            txtX.Text = tSelInfo.RCLeft.ToString();
            txtY.Text = tSelInfo.RCTop.ToString();
            txtWidth.Text = tSelInfo.RCWidth.ToString();
            txtHeight.Text = tSelInfo.RCHeight.ToString();

            //PerformRegionCapture(hwnd, x1, y1, width, height);
            this.WindowState = FormWindowState.Normal;
        }

        private void button1_Click(object sender, EventArgs e)
        {
            m_bRegOrWnd = false;
            
            TSelectionInfo tSelInfo = null;

            this.WindowState = FormWindowState.Minimized;
            System.Threading.Thread.Sleep(100);

            try
            {              
                tSelInfo = (TSelectionInfo)m_tSelection.Start(
                            (int)TS_SELECTION.tsSelectionUIElement,
                            (int)TS_SELECTION_OPTIONS.tsSelFlagDefaultText
                        );
            }
            catch (Exception ex)
            {
                MessageBox.Show(ex.Message);
                return;
            }

            // get the hwnd
            int hwnd = tSelInfo.WindowHandle;
            txtHandle.Text = hwnd.ToString("X");

            // set the UI element ID
            UIElem uiElem = ComFactory.Instance.NewUIElem();

            if (tSelInfo.UIElementID == "")
            {
                textBox1.Text = "A valid ID could not be generated!";
                this.WindowState = FormWindowState.Normal;

                return;
            }
            try
            {
                uiElem.InitializeFromID(tSelInfo.UIElementID, true);
                string uiElemID = uiElem.GetID(false);
                textBox1.Text = uiElemID;
            }
            catch (Exception ex)
            {
                MessageBox.Show(ex.Message);
            }

            this.WindowState = FormWindowState.Normal;
        }

        private void btnRectangle_Click(object sender, EventArgs e)
        {
            int x = int.Parse(txtX.Text);
            int y = int.Parse(txtY.Text);
            int width = int.Parse(txtWidth.Text);
            int height = int.Parse(txtHeight.Text);

            UIElem uiElem = null;
            try
            {
                uiElem = ComFactory.Instance.NewUIElem();
            }
            catch (Exception ex)
            {
                MessageBox.Show(ex.Message);
                return;
            }

            if (checkIDorHWND.Checked == true)
            {
                // get id
                string strID = textBox1.Text;
                try
                {
                    uiElem.InitializeFromID(strID, true);
                    m_bCanTryAgain = true;
                    PerformRegionCapture(uiElem.hwnd, x, y, width, height);
                }
                catch (Exception ex)
                {
                    MessageBox.Show(ex.Message);
                    return;
                }
            }
            else
            {
                // get the hwnd
                int hwnd = int.Parse(txtHandle.Text, System.Globalization.NumberStyles.HexNumber);
                m_bCanTryAgain = true;
                PerformRegionCapture(hwnd, x, y, width, height);
            }
            
        }

        private void button2_Click(object sender, EventArgs e)
        {
            UIElem uiElem = null;
            try
            {
                uiElem = ComFactory.Instance.NewUIElem();
            }
            catch (Exception ex)
            {
                MessageBox.Show(ex.Message);
                return;
            }

            if (checkIDorHWND.Checked == true)
            {
                // get id
                string strID = textBox1.Text;
                try
                {
                    uiElem.InitializeFromID(strID, true);
                }
                catch (Exception ex)
                {
                    MessageBox.Show(ex.Message);
                    return;
                }
            }
            else
            {
                // get the hwnd
                int hwnd = int.Parse(txtHandle.Text, System.Globalization.NumberStyles.HexNumber);
                uiElem.hwnd = hwnd;
            }

            m_bCanTryAgain = true;
            PerformWndCapture(uiElem);
        }

        private void btnActive_Click(object sender, EventArgs e)
        {
            int activeHwnd;

            this.WindowState = FormWindowState.Minimized;
            try
            {
                activeHwnd = m_tCapture.GetActiveWindowHwnd();
                string strResult = m_tCapture.CaptureWindow(activeHwnd);
                txtResult.Text = strResult;
            }
            catch (Exception ex)
            {
                MessageBox.Show(ex.Message);
                txtResult.Text = "Capture Error!";
            }
            this.WindowState = FormWindowState.Normal;
        }

        private void btnPoint_Click(object sender, EventArgs e)
        {
            int x = int.Parse(txtX.Text);
            int y = int.Parse(txtY.Text);
            int hwnd = int.Parse(txtHandle.Text, System.Globalization.NumberStyles.HexNumber);

            try
            {
                string strResult = m_tCapture.GetTextFromPoint(hwnd, x, y);
                txtResult.Text = strResult;
            }
            catch (Exception ex)
            {
                MessageBox.Show(ex.Message);
            }
            
        }
    }
}
